multi-agent/adapters/inbound/flask/endpoints/resources.py

```
@@ -104,23 +140,105 @@ def update_resource(resource_id, config, name=None):
     except ResourceInUseError as e:
-        # The resource is referenced by blueprints or other resources
-        return jsonify({"error": str(e),
-                        "blueprints": e.by_blueprints,
-                        "resources": e.by_resources}), 400
+        return jsonify({
+            "error": str(e),
+            "code": "RESOURCE_IN_USE",
+            "category": resource.category,
+            **_enrich_usage(e.by_blueprints, e.by_resources),
+        }), 409
+    except Exception as e:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)